src/test_l3_dsFPD.c

```diff
@@ -0,0 +1,627 @@
+}
+/**
+* @brief This test provides a scope to get Front panel LED Brightness.
+*
+* This test case provides a scope to  get the brightness of given Indicator.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+* Front panel idicator state should be set to On through Test 2 before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 6 and provide indicator index.
+*
+*/
+void test_6_dsFPD_hal_GetFPBrightness(void)
+{
+    gTestID = 6;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    dsFPDBrightness_t brightness = 0;
+
+    selectIndicator();
+    UT_LOG_INFO("Calling dsGetFPBrightness(IN:Indicator:[0x%d], OUT:Brightness)" \
+                                                ,eIndicator);
+    status = dsGetFPBrightness((dsFPDIndicator_t)eIndicator,&brightness);
+    UT_LOG_INFO("Result dsGetFPState(IN:Indicator:[0x%d], OUT:Brightness:[%d]) dsError_t:[%s]" \
+                                                ,eIndicator,brightness, UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+    assert(status == dsERR_NONE);
+    /* Check that the Indicator is valid */
+
+    UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to set Power LED state.
+*
+* This test case provides a scope to  set the Power LED to differten FP States.
+*
+* **Pre-Conditions:** @n
+* Front panel Module should be intialized through Test 1 before calling this test.
+* Front panel indicator state should be set to ON through Test 2 before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 7 and provide the FP state to check.
+*
+*/
+void test_7_dsFPD_hal_SetLEDState(void)
+{
+    gTestID = 7;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    uint32_t uLedState;
+    uint32_t uSupportedLEDState= 0;
+
+    UT_LOG_MENU_INFO(" \t  Supported POWER LED States are:");
+    UT_LOG_MENU_INFO("------------------------------------------");
+    uSupportedLEDState = UT_KVP_PROFILE_GET_UINT32("dsFPD/SupportedLEDStates");
+
+    for (int fpState = dsFPD_LED_DEVICE_NONE; fpState <= dsFPD_LED_DEVICE_MAX; fpState++)
+    {
+        if (!(uSupportedLEDState & (1 << fpState)))
+            continue;
+        UT_LOG_MENU_INFO("\t%d.  %-20s\n", fpState, UT_Control_GetMapString(dsFrontPanelLEDState, fpState));
+    }
+    UT_LOG_MENU_INFO("----------------------------------------------------------");
+    UT_LOG_MENU_INFO("Select State: ");
+    scanf("%X", &uLedState);
+    readAndDiscardRestOfLine(stdin);
+
+
+    /* Check that the Indicator is valid */
+    UT_LOG_INFO("Calling dsFPSetLEDState(IN state:[0x%X])",uLedState);
+    status = dsFPSetLEDState((dsFPDLedState_t)uLedState);
```

Comment:
This is a bit tricky to validate different states as they differ from platform to platform in behavior.  Raise this point in tomorrows call, so that we can discuss.

Reply:
Need to discuss in the Scrum call